messages.md

```diff
@@ -0,0 +1,382 @@
+watchtower. The wallet must wait for the tower's `sig_ack` before sharing its signature
+for the unvault transaction.
+
+```json
+{
+    "method": "sig",
+    "params": {
+        "signature": "ALL|ANYONECANPAY Bitcoin ECDSA signature as hex",
+        "id": "tx uid"
+    }
+}
+```
+
+The `tx uid` is `sha256(txid)`.
+
+
+#### `sig_ack`
+
+The watchtower must not send an ACK if it did not build and checked the transactions, or
+if it is unable to bump the feerate with its current utxos.
+FIXME: What is a reasonable limit for a WT to not send the ACK because of insufficient
+funds ?
+
+```json
+{
+    "result": {
+        "ack": true,
+        "vault_id": "vault_uid"
+    }
+}
+```
+
+The `vault_uid` is `sha256(vault txid)`.
+
+
+#### `get_spend_requests`
+
+Regularly sent by a watchtower to the synchronisation server to learn about spending
+attempts.
+
+```json
+{
+    "method": "get_spend_requests",
+    "params": {}
+}
+```
+
+
+#### `spend_requests`
+
+The response to a `get_spend_requests`. Returns an arbitrarily-sized (can be 0) array of
+objects detailing a spending request.
+
+```json
+{
+    "result": {
+        "requests": [
+            {
+                "transaction": "fully signed spend transaction",
+                "timestamp": 0000000,
+                "vault_id": "vault_uid"
+            }
+        ]
+    }
+}
+```
+
+The `vault_uid` is `sha256(vault txid)`.
+
+
+#### `spend_opinion`
+
+Sent by a watchtower to the synchronisation server to signal its acceptance or refusal of
+a specific spend. The `reason` field must be set if accept is `false`, otherwise it's
+ignored by the sync server.
+
+FIXME: should we require a signature here ? Seems to be a belt-and-suspenders check since
+there is the enrypted-authenticated message transport protocol.
+
+FIXMENIT: this probably needs a better name :).
+
+```json
+{
+    "method": "spend_opinion",
+    "params": {
+        "vault_id": "vault_uid",
+        "accept": true,
+        "reason": ""
+    }
+}
+```
+
+The `vault_uid` is `sha256(vault txid)`.
+
+
+#### `request_spend`
+
+Sent by an authorized spender to signal their willingness to spend a vault.
+
+We use a timestamp as watchtowers might accept the same spending attempt in the future.
+
+```json
+{
+    "method": "request_spend",
+    "params": {
+        "transaction": "fully signed spend transaction",
+        "timestamp": 0000000,
+        "vault_id": "vault_uid"
+    }
+}
+```
+
+The `vault_uid` is `sha256(vault txid)`.
+
+
+#### `get_spend_opinions`
+
+Sent by an authorized spender when polling for watchtowers agreement regarding the spend
+attempt identified by `vault_id`.
+
+```json
+{
+    "method": "get_spend_opinions",
+    "params": {
+        "vault_id": "vault_uid"
+    }
+}
+```
+
+The `vault_uid` is `sha256(vault txid)`.
+
+
+#### `spend_opinions`
+
+The response to `get_spend_opinions`, an arbitrarily-sized (can be 0 if no watchtower
+responded) array of the response of each watchtower.
```

Comment:
for precision, sync server may have a label or an ID for each watchtower, or is this obfuscated on purpose ?
```json
"opinions": [
  {
   "watchtower_id": ":id",
   "accepted": true,
   "reason": ""
}
```

Reply:
It could. Maybe in a follow-up?

Reply:
Isn't the ID implicit in the authenticated communication?

Reply:
Yes, but for the sync server not the wallets.

Reply:
Why do the wallets need to know about the identity of a watchtower?

Reply:
I'd say for helping the authorized spenders to figure out which watchtower is blocking operations and might be misconfigured.